src/datomish/query/clauses.cljc

```diff
@@ -62,6 +62,12 @@
                           (attribute-in-source (:source cc) value)
                           (constant-in-source (:source cc) value))]))
 
+(defn merge-ccs [left right]
+  (assoc left
+         :from (concat (:from left) (:from right))
```

Comment:
Each one is a unique table alias, so we just use a sequence. They're pairs, `[[:datoms datoms123], [:datoms datoms124]]`, etc.